Firestore/Source/Remote/FSTRemoteStore.mm

```diff
@@ -198,7 +201,7 @@ - (void)disableNetworkInternal {
     [self cleanUpWriteStreamState];
 
     self.writeStream = nil;
-    self.watchStream = nil;
+    //self.watchStream = nil;
```

Comment:
So in today's topology of objects `FSTStream` represents the RPC. Once the RPC is finished it should be recreated. Unfortunately in the Objective-C gRPC object model there's no explicit channel object so if you mirror that into the C++ implementation you'll have to tear everything down.

The intent was always that FSTDatastore would hold the connection state and that streams (and other RPCs like commit and lookup) could share it.

So I'd recommend putting the stub, channel, and completion queue in the Datastore class and then creating the streams fresh as we do today.

Note that we don't want to reuse stream objects between calls specifically because they're the targets of callbacks from the network thread in Objective-C. When the client initiates stream teardown for whatever reason we want to avoid generating events into the system even though gRPC might still consider that object alive.

We'll have to find some way to similarly prevent additional events from bleeding through in this implementation.